src/main/java/com/example/training/controller/UserController.java

```diff
@@ -0,0 +1,47 @@
+package com.example.training.controller;
+
+import com.example.training.model.User;
+import com.example.training.service.UserServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+
+@RestController
+@RequestMapping("/users")
+@RequiredArgsConstructor
+public class UserController {
+    private final UserServiceImpl userService;
+
+    @GetMapping("{id}")
+    public ResponseEntity<User> getUserById(@PathVariable Integer id) {
+        return ResponseEntity.ok(userService.getById(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<HttpStatus> createUser(@RequestBody User user) {
+        userService.save(user);
+        return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+
+    @GetMapping
+    public List<User> getAll() {
+        return userService.getAll();
+    }
+
+    @PutMapping("{id}")
+    public ResponseEntity<User> updateUser(@PathVariable Integer id, @RequestBody User inputUser) {
+        return ResponseEntity.ok(userService.update(id, inputUser));
```

Comment:
id can be part of a User object.
I suggest combining create and update method and calling it "put".